### DISCUSSION TITLE
Access incoming request from pydantic (during validation)

### DISCUSSION THREAD

**Discussion by david-shiko** [fastapi/fastapi#3166] [Questions]
Is there a way to access the token from `schemas.py` file during pydantic validation? I want to decline an updating of sensitive fields (email or password) if the token not containing a special flag.
I can do it after pydantic validation, but I have multiple routers for every role, so this will be required to modify all `path` updating endpoints in these routers instead of modifying only one file.

**Comment by ghandic** [comment]
Could you post an example snippet, I'm struggling to understand what the intent is @david-shiko ? If you want to block a user from updating the database if they don't have access then you can use dependency injection on the route?

**Comment by tiangolo** [comment]
It would probably be better to ask in a GitHub issue, following the template that will guide you to structure the question in a way that makes it easier to understand your intention and need, to be able to help you better. And also providing a self-contained, [minimal, reproducible, example](https://stackoverflow.com/help/minimal-reproducible-example) that I can copy-paste to replicate it.

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
